modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/WebMessagingJavaClientCodegen.java

```diff
@@ -0,0 +1,75 @@
+package io.swagger.codegen.languages;
+
+import io.swagger.codegen.SupportingFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WebMessagingJavaClientCodegen extends PureCloudJavaClientCodegen {
+    protected Logger LOGGER = LoggerFactory.getLogger(WebMessagingJavaClientCodegen.class);
+
+    public WebMessagingJavaClientCodegen() {
+        super();
+    }
+
+    @Override
+    public String getName() {
+        return "webmessagingjava";
+    }
+
+    @Override
+    public void processOpts() {
+        super.processOpts();
+
+        // Wipe out all the templates so we can start with a clean slate
+        this.apiTemplateFiles.clear();
+        this.apiDocTemplateFiles.clear();
+        this.apiTestTemplateFiles.clear();
+        this.modelTemplateFiles.clear();
+        this.modelDocTemplateFiles.clear();
+        this.modelTestTemplateFiles.clear();
+        this.operationTemplateFiles.clear();
+        this.supportingFiles.clear();
+
+        modelTemplateFiles.put("model.mustache", ".java");
+        modelDocTemplateFiles.put("model_doc.mustache", ".md");
+        apiTemplateFiles.put("api.mustache", ".java");
+        apiTemplateFiles.put("api_async.mustache", "Async.java");
+        operationTemplateFiles.put("requestBuilder.mustache", ".java");
+
+        supportingFiles.add(new SupportingFile("pom.mustache", "", "pom.xml"));
+        supportingFiles.add(new SupportingFile("gitignore.mustache", "", ".gitignore"));
+        supportingFiles.add(new SupportingFile("README.mustache", "", "README.md"));
+
+        final String invokerFolder = (sourceFolder + '/' + invokerPackage).replace(".", "/");
+        supportingFiles.add(new SupportingFile("WebMessagingClient.mustache", invokerFolder, "WebMessagingClient.java"));
+        supportingFiles.add(new SupportingFile("WebMessagingException.mustache", invokerFolder, "WebMessagingException.java"));
+        supportingFiles.add(new SupportingFile("GenesysCloudRegionHosts.mustache", invokerFolder, "GenesysCloudRegionHosts.java"));
+        supportingFiles.add(new SupportingFile("GenesysCloudRegionWebSocketHosts.mustache", invokerFolder, "GenesysCloudRegionWebSocketHosts.java"));
+        supportingFiles.add(new SupportingFile("ApiDateFormat.mustache", invokerFolder, "ApiDateFormat.java"));
+        supportingFiles.add(new SupportingFile("LocalDateDeserializer.mustache", invokerFolder, "LocalDateDeserializer.java"));
+        supportingFiles.add(new SupportingFile("Configuration.mustache", invokerFolder, "Configuration.java"));
+        supportingFiles.add(new SupportingFile("Pair.mustache", invokerFolder, "Pair.java"));
+        supportingFiles.add(new SupportingFile("StringUtil.mustache", invokerFolder, "StringUtil.java"));
+
+        final String authFolder = (sourceFolder + '/' + invokerPackage + ".auth").replace(".", "/");
+        supportingFiles.add(new SupportingFile("auth/HttpBasicAuth.mustache", authFolder, "HttpBasicAuth.java"));
```

Comment:
We shouldn't need any of the built-in auth templates for this. The ApiClient instance should only have a simple way to set the token obtained from the websocket. These auth helpers are only useful when generating against arbitrary swagger definitions that may use any of the swagger-supported auth mechanisms.

Reply:
I've gone ahead and removed all except OAuth.mustache and Authentication.mustache because they're needed when setting the access token manually. I can remove them as well if you'd like but there'd be changes needed in ApiClient to handle that.